### DISCUSSION TITLE
How do I disable mini.indentline in the dashboard

### DISCUSSION THREAD

**Discussion by cjnucette** [folke/snacks.nvim#157] [Q&A] [answered]
I am trying the following:
```lua
vim.api.nvim_create_autocmd('Filetype', {
	desc = "Disable mini.indentline by filetype",
	group = vim.api.nvim_create_augroup('disable-miniindentline', { clear = true }),
	pattern = { 'snacks_dashboard' },
	callback = function(args)
		vim.b[args.buf].miniindentscope_disable = true
	end
})
```
but the pattern doesn't seem to match.

**Comment by tsakirist** [comment]
I want to disable other plugins as well there but it seems that the filetype="snacks_dashboard" is not emitted when initially started. If you open it manually with `Snacks.dashboard.open()` the event is emitted and the aucmds work.

**Reply by dpetka2001** [reply]
If you're using `lazy.nvim` just lazy-load `mini.indentscope` in `event = BufReadPost` or something. Then it shouldn't be loaded in Snacks dashboard on startup.

**Reply by tsakirist** [reply]
That's what I've done for that already, albeit not ideal. There could be cases where one might need the `ft` (?) so if my _assumption_ is correct and the ft is not emitted initially could be problematic.

**Reply by cjnucette** [reply]
@dpetka2001 Thanks for the suggestion, but I would rather see this fixed.

**Reply by dpetka2001** [reply]
It's not about the filetype being emitted or not. Snacks dashboard loads immediately at startup and the problem is that the `FileType` event from Neovim does not get fired until afterwards. So, the autocmd on `FileType` doesn't get evaluated on startup. `FileType` event usually happens after a buffer is read. It's a matter of how Neovim events happen and trigger.

**Reply by dpetka2001** [reply]
@cjnucette I don't believe there's anything to be fixed since Neovim itself does not trigger the `FileType` event until afterwards. I have the following script in my dotfiles (which I borrowed either from [here](https://github.com/jdhao/nvim-config/blob/d86f5994833cc3a108c81787f788fbc76108cdca/plugin/log-autocmds.vim) or [here](https://github.com/pysan3/dotfiles/blob/3323fd39150b76b07f2c790d17f912ac89da881c/static/log-autocmds.vim)). I put that script in `~/.config/nvim/plugin/log-autocmds.vim` in my personal configuration.

Then just run `nvim -c "LogAutocmds"` and after exiting Neovim rename `/tmp/vim_log_autocommands` to `/tmp/vim_log_autocommands_no_file`. Then run again `nvim -c "LogAutocmds"` and press `s` to restore a session and exit again. Neovim will feel a little sluggish during these steps because of the logging of the events happening in the background. If you then diff `/tmp/vim_log_autocommands` with `/tmp/vim_log_autocommands_no_file` you will see that in the file `/tmp/vim_log_autocommands_no_file` no `FileType` event has been triggered. 

It's just the way Neovim triggers different events in different order. When dashboards load at startup, the `FileType` event has not yet been triggered by Neovim, so autocmds which happen on `FileType` won't get evaluated. `FileType` happens after `BufReadPre` which happens just before a file is read into the buffer.

**Comment by infinite-ops** [comment, marked as answer]
I did this which looks to work for me
```
vim.api.nvim_create_autocmd("User", {
  pattern = "SnacksDashboardOpened",
  callback = function(data)
    vim.b[data.buf].miniindentscope_disable = true
  end,
})
```

**Reply by dpetka2001** [reply]
Yes, that's the way. You have to use snacks dashboard's internal events, since `FileType` won't get triggered at startup. Still the same can be achieved by simply lazy-loading the plugin. I don't understand the difference to be honest.

**Comment by cjnucette** [comment]
@infinite-ops That is it. Thanks.

**Comment by geodimm** [comment]
A bit late to the party but if you're using lazy.nvim as a plugin manager you can also do

```lua
{
  'folke/snacks.nvim',
  init = function()
    vim.b.miniindentscope_disable = true
  end,
  opts = {
  }
}
```

This is exactly what the init function is useful for as per the spec: https://lazy.folke.io/spec#spec-setup
